packages/flutter/lib/src/services/raw_keyboard_web.dart

```diff
@@ -0,0 +1,191 @@
+  /// for more information.
+  final int keyCode;
+
+  /// The modifiers that were present when the key event occurred.
+  ///
+  /// See `lib/src/engine/keyboard.dart` in the web engine for the numerical
+  /// values of the `metaState`. These constants are also replicated as static
+  /// constants in this class.
+  ///
+  /// See also:
+  ///
+  ///  * [modifiersPressed], which returns a Map of currently pressed modifiers
+  ///    and their keyboard side.
+  ///  * [isModifierPressed], to see if a specific modifier is pressed.
+  ///  * [isControlPressed], to see if a CTRL key is pressed.
+  ///  * [isShiftPressed], to see if a SHIFT key is pressed.
+  ///  * [isAltPressed], to see if an ALT key is pressed.
+  ///  * [isMetaPressed], to see if a META key is pressed.
+  final int metaState;
+
+  @override
+  final String keyLabel;
+
+  @override
+  PhysicalKeyboardKey get physicalKey {
+    if (kWebToPhysicalKey.containsKey(keyCode)) {
+      return kWebToPhysicalKey[keyCode] ?? PhysicalKeyboardKey.none;
+    }
+    return PhysicalKeyboardKey.none;
+  }
+
+  @override
+  LogicalKeyboardKey get logicalKey {
+    // Look to see if the keyCode is a printable number pad key, so that a
+    // difference between regular keys (e.g. ".") and the number pad version
+    // (e.g. the "." on the number pad) can be determined.
+    final LogicalKeyboardKey numPadKey = kWebNumPadMap[keyCode];
+    if (numPadKey != null) {
+      return numPadKey;
+    }
+
+    // Look to see if the keyCode is one we know about and have a mapping for.
+    final LogicalKeyboardKey newKey = kWebToLogicalKey[keyCode];
+    if (newKey != null) {
+      return newKey;
+    }
+
+    // This is a non-printable key that we don't know about, so we mint a new
+    // code with the autogenerated bit set.
+    const int webKeyIdPlane = 0x00900000000;
+    return LogicalKeyboardKey(
+      webKeyIdPlane | keyCode | LogicalKeyboardKey.autogeneratedMask,
+      debugName: kReleaseMode ? null : 'Unknown Web key code $keyCode',
+    );
+  }
+
+  @override
+  bool isModifierPressed(
+    ModifierKey key, {
+    KeyboardSide side = KeyboardSide.any,
+  }) {
+    switch (key) {
+      case ModifierKey.controlModifier:
+        return metaState & modifierControl != 0;
+      case ModifierKey.shiftModifier:
+        return metaState & modifierShift != 0;
+      case ModifierKey.altModifier:
+        return metaState & modifierAlt != 0;
+      case ModifierKey.metaModifier:
+        return metaState & modifierMeta != 0;
+      case ModifierKey.numLockModifier:
+        return metaState & modifierNumLock != 0;
+      case ModifierKey.capsLockModifier:
+        return metaState & modifierCapsLock != 0;
+      case ModifierKey.scrollLockModifier:
+        return metaState & modifierScrollLock != 0;
+      case ModifierKey.functionModifier:
+      case ModifierKey.symbolModifier:
+        // On Web, the browser doesn't report the state of the FN and SYM modifiers.
+        return false;
```

Comment:
Can you have `default` together with fucntionModifier and symbolModifier and avoid two returns?